blocks/search-results/search-results.js

```diff
@@ -0,0 +1,84 @@
+class SearchResults {
+  constructor(element, options) {
+    this.element = element;
+    this.options = options;
+  }
+
+  init() {
+    this.searchterm = SearchResults.getSearchTerm();
+
+    if (this.searchterm) {
+      this.searchterm.trim();
+      this.getResults();
+    }
+  }
+
+  getResults() {
+    fetch(this.options.endpoint, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        query: `
+        query ($q: String) {search(q: $q){
+              count
+            items {
+              title
+              snippet
+              path
+            }
+          }
+        }
+      `,
+        variables: {
+          q: this.searchterm,
+        },
+      }),
+    })
+      .then((res) => res.json())
+      .then((result) => {
+        this.searchResults = result.data.search;
+        this.showResults();
+      })
+      .catch(() => {
+        this.showResults();
+      });
+  }
+
+  showResults() {
+    let resultsText = window.placeholders?.default?.noResultsFor.replace('{value}', this.searchterm) || `No results for ${this.searchterm}`;
```

Comment:
not sure if this is a standard but I am used to i18n placeholders like `{}` or `{0}`, should we use the same?